bundles/binding/org.openhab.binding.panstamp/src/main/java/org/openhab/binding/panstamp/internal/PanStampBinding.java

```diff
@@ -0,0 +1,566 @@
+			setupDirectories(cfg);
+			network.setDeviceStore(store);
+			logger.info("Configured binding product codes");
+			network.addListener(networkListener);
+			network.open();
+			logger.info("Opened network");
+//			setProperlyConfigured(true);
+		} catch (NetworkException e) {
+			logger.error("Error connecting to serial network on {}: {}",
+					cfg.serialPort, e.getMessage());
+
+		} catch (SwapException e) {
+			logger.error(
+					"Error reading modem settings for serial network on {}: {}",
+					cfg.serialPort, e.getMessage());
+		} catch (Throwable e) {
+			logger.error("Fatal error: {}", e.getMessage(), e);
+		}
+	}
+
+	private void stopNetwork() {
+		logger.debug("stopNetwork()");
+		try {
+			network.close();
+			network = null;
+			logger.info("Closed panStamp network");
+		} catch (ModemException e) {
+			logger.error("Error closing panStamp network: " + e.getMessage(), e);
+		} finally {
+//			setProperlyConfigured(false);
+		}
+	}
+
+	private void startDebug(final PanStampBindingSettings cfg) {
+		if (cfg.debugEnabled) {
+			try {
+				debugServer = new TcpServer(network.getSWAPModem(),
+						cfg.debugPort);
+				logger.info("TCP Debug enabled on port {}", cfg.debugPort);
+			} catch (TcpException e) {
+				logger.error("Error creating debug service:  {}"
+						+ e.getMessage());
+			}
+		}
+	}
+
+	private void stopDebug() {
+		try {
+			if (debugServer != null) {
+				debugServer.close();
+				debugServer = null;
+			}
+		} catch (TcpException e) {
+			logger.error("Error closing debug port: " + e.getMessage(), e);
+		}
+	}
+
+	private void setupModem(PanStampBindingSettings cfg) {
+		try {
+			ModemSetup setup = modem.getSetup();
+			if (cfg.networkId != -1) {
+				setup.setNetworkID(cfg.networkId);
+			}
+			if (cfg.networkDeviceAddress != -1) {
+				setup.setDeviceAddress(cfg.networkDeviceAddress);
+			}
+			if (cfg.networkChannel != -1) {
+				setup.setChannel(cfg.networkChannel);
+			}
+			if (cfg.networkSecurityOption != -1) {
+				// FIXME
+			}
+			if ((cfg.networkId != -1) || (cfg.networkChannel != -1)
+					|| (cfg.networkDeviceAddress != -1)) {
+				modem.setSetup(setup);
+				logger.info(
+						"Modem setup updated to address {}, channel {}, and network ID {}",
+						setup.getDeviceAddress(), setup.getChannel(),
+						setup.getNetworkID());
+			}
+		} catch (SwapException e) {
+			logger.error("Error configuring network: {}", e.getMessage());
+		}
+	}
+
+	private void setupDirectories(PanStampBindingSettings cfg) {
+		File xmlDir = new File(cfg.xmlDir);
+		if (xmlDir.exists() && xmlDir.isDirectory() && xmlDir.canRead()) {
+			File devFile = new File(xmlDir, "devices.xml");
+			if (devFile.exists() && devFile.canRead() && devFile.isFile()) {
+				network.setDeviceLibrary(new CompoundDeviceLibrary(
+						new FileLibrary(xmlDir), new ClassLoaderLibrary()));
+				logger.info("Configured XML directory to {}", cfg.xmlDir);
+			} else {
+				logger.error(
+						"Devices file in XML directory {} can not be read",
```

Comment:
Change "Devices" to "devices.xml" so user knows specifically the file name that cannot be read.